src/main/java/org/terasology/web/serverAdminManagement/IdPermissionPair.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.serverAdminManagement;
+
+import java.util.Objects;
+
+/**
+ * This class is a simple container that follows the pair pattern.
+ * @param <T> the first part of the pair.
+ * @param <U> the second part of the pair.
+ */
+public class IdPermissionPair<T, U> {
```

Comment:
I think this class is attempting to mix two goals. What should it actually represent?
- If it should model a generic pair, like the one from JavaFX, using the T and U type parameters is correct but I think the names of the two fields shouldn't refer to a specific usage (`id` and `permissions`), but should be something like `first` and `second` (like [C++'s pair](http://www.cplusplus.com/reference/utility/pair/)), or `key` and `value` (like [JavaFX's pair](https://docs.oracle.com/javase/8/javafx/api/javafx/util/Pair.html) as you already know), or `left` and `right`, etc. In this case, I'd also move the class to a different package (maybe there is an `utils`/`utilities` package somewhere, or you may create one at the root of the project) since it can also be used for other purposes which aren't related to server admin management 
- If you want it to only model an *id-permissions pair*, then I'd keep these names for the fields but I wouldn't use generics - an id is always a `String` and a set of permissions is always a `Map<PermissionType, Boolean>`. In this case I think the class is already in the correct package, since it's only used for admin management purposes